pkg/ui/ccl/src/views/clusterviz/containers/map/modalLocalities.tsx

```diff
@@ -6,106 +6,143 @@
 import { NodeView } from "./nodeView";
 import { Box, ZoomTransformer } from "./zoom";
 
-type BoxDimensions = {x: number, y: number, w: number, h: number};
-
 interface LocalityViewProps {
-  locality: LocalityTreeNode;
+  locality: LocalityTree;
 }
 
 class LocalityView extends React.Component<LocalityViewProps, any> {
+  static contextTypes = {
+    router: PropTypes.object.isRequired,
+  };
+  context: { router: InjectedRouter & RouterState };
+
+  onClick = () => {
+    const tiers = this.props.locality.tiers.map(({ key, value }) => key + "=" + value).join("/");
+    const destination = "/clusterviz/" + tiers;
+    this.context.router.push(destination);
+  }
+
   render() {
+    const { tiers } = this.props.locality;
+    const thisTier = tiers[tiers.length - 1];
+
     return (
-      <text x={15} y={15}>
+      <text x={15} y={15} onClick={this.onClick} style={{ cursor: "pointer" }}>
         {
-          // TODO Resolve name of locality.
-          (this.props.locality.data as any).key
+          thisTier.key + "=" + thisTier.value
         }
       </text>
     );
   }
 }
 
 interface LocalityBoxProps {
-  box: Box;
-  data: LocalityTreeNode;
+  projection: d3.geo.Projection;
+  node: LocalityTree;
+  locationTree: LocationTree;
 }
 
 class LocalityBox extends React.Component<LocalityBoxProps, any> {
   render() {
+    const location = findMostSpecificLocation(this.props.locationTree, this.props.node.tiers);
+    const center = this.props.projection([location.longitude, location.latitude]);
+    const box = new Box(center[0] - 50, center[1] - 50, 100, 100);
     return (
-      <g transform={`translate(${this.props.box.center()})`}>
-        <LocalityView locality={this.props.data} />
+      <g transform={`translate(${box.center()})`}>
+        <LocalityView locality={this.props.node} />
       </g>
     );
   }
 }
 
 interface NodeBoxProps {
-  box: Box;
-  data: SimulatedNodeStatus;
+  projection: d3.geo.Projection;
+  node: NodeStatus$Properties;
+  nodeHistory: SimulatedNodeStatus;
+  locationTree: LocationTree;
 }
 
 class NodeBox extends React.Component<NodeBoxProps, any> {
   render() {
+    const tiers = this.props.node.desc.locality.tiers.map(({ key, value }) => ({ key, value }));
+    const location = findMostSpecificLocation(this.props.locationTree, tiers);
+    const center = this.props.projection([location.longitude, location.latitude]);
+    const box = new Box(center[0] - 50, center[1] - 50, 100, 100);
     return (
-      <g transform={`translate(${this.props.box.center()})`}>
+      <g transform={`translate(${box.center()})`}>
         <NodeView
-          nodeHistory={this.props.data}
+          nodeHistory={this.props.nodeHistory}
           maxClientActivityRate={10000}
         />
       </g>
     );
   }
 }
 
-interface ModalTreeNodeViewProps {
-  node: LocalityTreeNode;
+interface ModalLocalitiesViewProps {
+  localityTree: LocalityTree;
+  locationTree: LocationTree;
+  tiers: LocalityTier[];
```

Comment:
naming quibble: would prefer to call this localityPath or something, because it's the path we've walked down in the locality tree.

I understand that you're trying to match how nodes have "tiers" on them though.